runners/samza/src/main/java/org/apache/beam/runners/samza/translation/TranslationContext.java

```diff
@@ -159,12 +182,29 @@ public MessageStream<OpMessage<String>> getDummyStream() {
   }
 
   public <OutT> MessageStream<OpMessage<OutT>> getMessageStream(PValue pvalue) {
+    return getMessageStream(pvalue, true);
+  }
+
+  public <OutT> MessageStream<OpMessage<OutT>> getMessageStream(
```

Comment:
There are some article about "adding a boolean parameter to change behavior".

Under this assumption, `getMessageStream(PValue pvalue)` and `getMessageStreamWithTransformMetric(PValue pvalue)` would be the better naming and signature.




https://medium.com/@amlcurran/clean-code-the-curse-of-a-boolean-parameter-c237a830b7a3
https://alexkondov.com/should-you-pass-boolean-to-functions/

Reply:
yea i was already wondering if there is cleaner way to do this, but seems ok here since there are only limited number of translators we will have?

Reply:
I don't think it's okay. Once we add a boolean flag, it will open the path two add more later. The number of code paths in one method is 2^n.

I would prefer

```
public <OutT> MessageStream<OpMessage<OutT>> getMessageStreamWithTransformMetric(PValue pvalue) {
    MessageStream<OpMessage<OutT>> stream = getMessageStream(pvalue);
        final Config overrideConfig = new MapConfig(getPipelineOptions().getConfigOverride());
    if (doAttachMetricOp(overrideConfig)) {
      // add another step if registered for Op Stream
      stream.flatMapAsync(
          OpAdapter.adapt(
              new SamzaInputMetricOp<>(
                  pvalue.getName(), getTransformFullName(), beamTransformMetricRegistry),
              this));
    }
    return stream;
}
```

or 

```
public <OutT> MessageStream<OpMessage<OutT>> addTransformMetric(OpMessage<OutT> stream) {
...
}
```

and use the 2 combined.

Reply:
check the refactor